tests/components/bmw_connected_drive/test_coordinator.py

```diff
@@ -134,32 +190,34 @@ async def test_init_reauth(
 async def test_captcha_reauth(
     hass: HomeAssistant,
     freezer: FrozenDateTimeFactory,
+    issue_registry: ir.IssueRegistry,
 ) -> None:
-    """Test the reauth form."""
-    TEST_REGION = "north_america"
-
-    config_entry_fixure = deepcopy(FIXTURE_CONFIG_ENTRY)
-    config_entry_fixure["data"][CONF_REGION] = TEST_REGION
-    config_entry = MockConfigEntry(**config_entry_fixure)
+    """Test a CaptchaError initializing reauth flow."""
+    config_entry = MockConfigEntry(**FIXTURE_CONFIG_ENTRY)
     config_entry.add_to_hass(hass)
 
     await hass.config_entries.async_setup(config_entry.entry_id)
     await hass.async_block_till_done()
 
-    coordinator = config_entry.runtime_data
-
-    assert coordinator.last_update_success is True
+    # Test if entities show data correctly
+    for entity_id, state in FIXTURE_ENTITY_STATES.items():
+        assert hass.states.get(entity_id).state == state
 
-    freezer.tick(timedelta(minutes=10, seconds=1))
+    # If library decides a captcha is needed, we should initialize reauth flow
+    freezer.tick(timedelta(minutes=5, seconds=1))
     with patch(
-        "bimmer_connected.account.MyBMWAccount.get_vehicles",
-        side_effect=MyBMWCaptchaMissingError(
-            "Missing hCaptcha token for North America login"
-        ),
+        BIMMER_CONNECTED_VEHICLE_PATCH,
+        side_effect=MyBMWCaptchaMissingError("Missing hCaptcha token"),
     ):
         async_fire_time_changed(hass)
         await hass.async_block_till_done()
 
-    assert coordinator.last_update_success is False
-    assert isinstance(coordinator.last_exception, ConfigEntryAuthFailed) is True
-    assert coordinator.last_exception.translation_key == "missing_captcha"
+    for entity_id in FIXTURE_ENTITY_STATES:
+        assert hass.states.get(entity_id).state == "unavailable"
+    assert len(issue_registry.issues) == 1
+
+    reauth_issue = issue_registry.async_get_issue(
+        HOMEASSISTANT_DOMAIN,
+        f"config_entry_reauth_{BMW_DOMAIN}_{config_entry.entry_id}",
+    )
+    assert reauth_issue.active is True
```

Comment:
Is this to test we need to reauth? I think we generally check if there is an active flow for that

Reply:
Yes. It was done like this in the tests I copied from. Have added testing for the config flows as well.